### PR TITLE
Button says "Send" instead of "Schedule" even if the schedule time is in the future MAILPOET-4163

### PR DESCRIPTION
**Pull request by samnajian**
### This PR fixed the issue with scheduling newsletters for future/past dates/times.

This PR initially makes some cleanup and then applies the fix as follows:
- Define NewsLetter type
- Define component props type
- Add acceptance test for past/future time

### Testing:
- Pick a newsletter and visit `Edit` page
- Visit the `Send` page by clicking `Next` on the the editor page
- Toggle the `schedule` option on
- Verify that the primary submit button this page reads `Send` for past dates/times and `Schedule` for future dates/times, note that time is calculated based on the time show in `Your website’s time is {time}`

[MAILPOET-4163]

[MAILPOET-4163]: https://mailpoet.atlassian.net/browse/MAILPOET-4163?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/assets/js/src/newsletters/models/newsletter.ts [hunk 1]
```diff
@@ -1,4 +1,20 @@
-type NewsLetterType = 'standard';
+export enum NEWSLETTER_TYPE {
+  STANDARD = 'standard',
+  AUTOMATIC = 'automatic',
+  WELCOME = 'welcome',
+  NOTIFICATION = 'notification',
+  NOTIFICATION_HISTORY = 'notification_history',
+  WC_TRANSACTIONAL = 'wc_transactional',
+  RE_ENGAGEMENT = 're_engagement'
+}
```
Comment (JanJakes) [inline mailpoet/assets/js/src/newsletters/models/newsletter.ts:9]: This is just a stylistic note but we should preserve the same style as for other enums. This means `enum NewsletterType` and enum fields `Standard`, `Automatic`, etc.

Screen Shot 2022-03-11 at 13 41 32
Screen Shot 2022-03-11 at 13 42 28

File: mailpoet/assets/js/src/date.ts [hunk 2]
```diff
@@ -178,5 +178,5 @@ export const MailPoetDate : {
 
     return convertedFormat.join('');
   },
-  isInFuture: (dateString: string): boolean => new Date(dateString).getTime() > Date.now(),
+  isInFuture: (dateString: string, currentTime: MomentInput): boolean => Moment(dateString).isAfter(currentTime, 's'),
```
Comment (JanJakes) [inline mailpoet/assets/js/src/date.ts:181]: So I guess we don't use this function in any other place?
Seems like `currentTime` could be optional. But if we don't use it elsewhere, maybe not too necessary.

**Review by JanJakes — COMMENTED** [review]
LGTM, just tiny notes.

**Review by JanJakes — APPROVED** [review]
Good job!

**Review by veljkho — APPROVED** [review]

**Pull request merged**